tasks.py

```diff
@@ -102,6 +103,26 @@ def yarn(c, cmd, pty: bool = False):
     c.run(f'cd "{path}" && {cmd}', pty=pty)
 
 
+def node_available(versions: bool = False):
+    """Checks if the frontend environment (ie node and yarn in bash) is available."""
+    def ret(val, val0=None, val1=None):
+        if versions:
+            return val, val0, val1
+        return val
+
+    def check(cmd):
+        try:
+            return str(subprocess.check_output([cmd], stderr=subprocess.STDOUT, shell=True), encoding='utf-8').strip()
+        except subprocess.CalledProcessError:
+            return None
+        except FileNotFoundError:
+            return None
+
+    yarn_version = check('yarn --version')
+    node_version = check('node --version')
+    return ret((not yarn_version or not node_version), node_version, yarn_version)
```

Comment:
I would not enforce yarn, I would rather show a message that you should install it via `npm i -g yarn`, that's not a big deal from my side if you already have node installed, and even if you don't want yarn, you can also use npm. You can also install it only project wide and run via npx. Maybe add a flag to download no matter if you have node.

Reply:
All the frontend commands expect yarn and our locking file is shipped for yarn so I thin enforcing it is the safer choice.

Reply:
Ok, so there will be no warning if you already have node but not yarn, it'll default to download the frontend. Maybe just add a print here?